baking/src/tezos_baking/tezos_setup_wizard.py

```diff
@@ -822,13 +846,79 @@ def import_baker_key(self):
                 else:
                     baker_set_up = True
 
+    def stake_tez(self):
+        def get_minimal_frozen_stake():
+            output = get_proc_output(
+                f"curl {self.config['node_rpc_endpoint']}/chains/main/blocks/head/context/constants"
+            ).stdout.decode("utf-8")
+            return json.loads(output)["minimal_frozen_stake"]
+
+        def get_staked_balance(pkh):
+            output = get_proc_output(
+                f"curl {self.config['node_rpc_endpoint']}/chains/main/blocks/head/context/contracts/{pkh}/staked_balance"
+            ).stdout.decode("utf-8")
+            return json.loads(output)
+
+        def get_adaptive_issuance_launch_cycle():
+            output = get_proc_output(
+                f"curl {self.config['node_rpc_endpoint']}/chains/main/blocks/head/context/adaptive_issuance_launch_cycle"
+            ).stdout.decode("utf-8")
+            return json.loads(output)
+
+        def get_current_cycle():
+            output = get_proc_output(
+                f"curl {self.config['node_rpc_endpoint']}/chains/main/blocks/head/metadata"
+            ).stdout.decode("utf-8")
+            return json.loads(output)["level_info"]["cycle"]
+
+        tezos_client_options = self.get_tezos_client_options()
+        baker_alias = self.config["baker_alias"]
+        baker_key_hash = self.config["baker_key_hash"]
+
+        adaptive_issuance_launch_cycle = get_adaptive_issuance_launch_cycle()
+
+        current_cycle = get_current_cycle()
+
+        if adaptive_issuance_launch_cycle is not None and int(current_cycle) >= int(
+            adaptive_issuance_launch_cycle
+        ):
+            minimal_frozen_stake = get_minimal_frozen_stake()
+
+            staked_balance = get_staked_balance(baker_key_hash)
+
+            if int(staked_balance) < int(minimal_frozen_stake):
+
+                self.query_step(
+                    get_stake_tez_query(staked_balance, minimal_frozen_stake)
+                )
+
+                print_and_log(f"Staking {self.config['stake_tez']}Tz...")
+
+                if self.check_ledger_use():
+                    ledger_app = "Wallet"
+                    print(f"Please open the Tezos {ledger_app} app on your ledger.")
+                    print(
+                        color(
+                            f"Waiting for the Tezos {ledger_app} to be opened...",
+                            color_green,
+                        ),
+                    )
```

Comment:
Could you please also add the following prompt here?

`If you are using Tezos Wallet app of version 3.0.0 or higher, you need to enable "expert mode" in the Tezos Wallet app settings on the Ledger device.`

This is the detailed introduced by recent `tezos-wallet` app release, and it's relevant only for `stake` operation, i.e. we don't need to add it to `tezos-vote` wizard